manifests/networkd.pp

```diff
@@ -0,0 +1,24 @@
+# **NOTE: THIS IS A [PRIVATE](https://github.com/puppetlabs/puppetlabs-stdlib#assert_private) CLASS**
+#
+# This class provides an abstract way to trigger resolved
+#
+# @param ensure
+#   The state that the ``networkd`` service should be in
+#
+class systemd::networkd (
+  Variant[Enum['stopped','running'],Boolean] $ensure = 'running',
```

Comment:
If it's a private class and the values are limited to `Enum['stopped','running']` in `init.pp`, I'm not sure how it could ever be a Boolean?